### ISSUE TITLE
Keep relations without players alive until the end of a transaction

### ISSUE THREAD

**Issue by lveillard** [vaticle/typedb#6920]
## Current behaviour:
- Unlinking the last player of any relation in any operation inside a transaction triggers a deletion of the relation
- Creating a relation without players for further edition in next streams is not allowed

## Expected behaviour:
- Relations should survive while being transformed inside the transaction.

## Example use case
RIght now some operations like creating a relation and linking it to N matched nodes in the same mutation is not possible.  There is no workaround, not possible in a single stream and not possible in multiple streams. (Detailed here #6902) 
```
$allPeople isa Person;
$company isa Company, has id 'vaticle'

insert
$r isa contract, has id 'new contract'
$r (employees: $allPeople, company: $company)
```
We get this error

```
Attempted to assign a key 'new contract' of type 'id' that had been taken by another 'contract'."
```

## Proposed solution
The  "quick" solution that seems not to have breaking changes would be to:
- Keep relations without players alive during a transaction
- Trigger deletion of orphan relations only on transaction commit

## Enabled use cases
This enables two use cases:
1) the example of the issue: `create then link` in two streams (no workaround)
2) replacing edges: `unlink then link` (which would enable batched replaces)

Thanks!

**Comment by flyingsilverfin** [comment]
Good news :) We decided to allow keeping relations alive without role players until the end of the transaction. We will roll this into the next big release (2.25 - fetch).

Note that in the future we will probably disallow using IID in queries. To work around this, you can add a temporary attribute/key to your relations you are mutating:
1. Find the relation you want to delete role players from, delete the role players, and add an attribute
2. Look up the relation by the new attribute and then add new role players.

Hope this unblocks you!

**Comment by flyingsilverfin** [comment]
Closed with #6927
